Wrappers/Python/cil/optimisation/algorithms/SIRT.py

```diff
@@ -97,22 +145,61 @@ def set_up(self, initial, operator, data, lower=None, upper=None, constraint=Non
                 self.constraint=IndicatorBox(lower=lower,upper=upper)
                 
         # Set up scaling matrices D and M.
-        self.M = 1/self.operator.direct(self.operator.domain_geometry().allocate(value=1.0))        
-        self.D = 1/self.operator.adjoint(self.operator.range_geometry().allocate(value=1.0))
+        self.M = 1./self.operator.direct(self.operator.domain_geometry().allocate(value=1.0))                
+        self.D = 1./self.operator.adjoint(self.operator.range_geometry().allocate(value=1.0))
+
         self.configured = True
-        print("{} configured".format(self.__class__.__name__, ))
+        logging.info("{} configured".format(self.__class__.__name__, ))
 
+    def fix_weights(self):
 
+        r""" In case of division error when the preconditioning arrays :code:`M` and :code:`D`
+        are defined, the :code:`np.nan`, :code:`+np.inf` and :code:`-np.inf` values are replaced with one.
+        """
+        
+        # fix for possible inf values (code from numpy.nan_to_nam)
+        np_version = numpy.__version__.split('.')
+        if np_version[0]=='1' and np_version[1]<='17':
```

Comment:
does the `<='17'` work correctly as `17` is a string? instead can we convert the elements of np_version to `int`

Reply:
Fixed